thavalon-server/src/game/state/on_mission.rs

```diff
@@ -118,6 +119,51 @@ impl GameState<OnMission> {
     fn includes_player(&self, player: &str) -> bool {
         self.proposal().players.contains(player)
     }
+
+    // Append messages for lovers to an exisiting list of effects. This will not modify effects if there are no lovers in game,
+    // so this is safe to call in any game.
+    fn add_lover_effects(&self, effects: &mut Vec<Effect>) {
+        let tristan_display_name = self.game.display_name_from_role(Role::Tristan);
+        let tristan_on_mission = match tristan_display_name {
+            Some(name) => Some(self.proposals.get(self.phase.proposal_index).unwrap().players.contains(name)),
+            None => None,
+        };
+        let iseult_display_name = self.game.display_name_from_role(Role::Iseult);
+        let iseult_on_mission = match iseult_display_name {
+            Some(name) => Some(self.proposals.get(self.phase.proposal_index).unwrap().players.contains(name)),
+            None => None,
+        };
+        // First, add messages going to Tristan. If there is no Tristan, do nothing.
+        if tristan_display_name.is_some() {
+            effects.push(self.build_lover_effect(&tristan_display_name.unwrap().as_ref(), tristan_on_mission.unwrap(), iseult_display_name, iseult_on_mission));
+        }
+        // Next do the same for Iseult.
+        if iseult_display_name.is_some() {
+            effects.push(self.build_lover_effect(&iseult_display_name.unwrap().as_ref(), iseult_on_mission.unwrap(), tristan_display_name, tristan_on_mission));
```

Comment:
The `display_name` variables contain references already, so I think just an `unwrap` here is sufficient. Rust will automatically dereference as needed, but I think this is passing in a &&&String that Rust then double dereferences to a &str.